client/src/pages/SignIn.js

```diff
@@ -1,45 +1,85 @@
-export default function SignIn() {
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+
+export default function SignIn({ onSignIn }) {
+  const [isLoading, setIsLoading] = useState(false);
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+    try {
+      setIsLoading(true);
+      const formData = new FormData(event.target);
+      const userData = Object.fromEntries(formData.entries());
+      const req = {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(userData),
+      };
+      const res = await fetch('/api/auth/sign-in', req);
+      if (!res.ok) {
+        throw new Error(`fetch Error ${res.status}`);
+      }
+      const { user, token } = await res.json();
+      onSignIn();
+
+      localStorage.setItem('token', token);
```

Comment:
You may find it convenient later to also store the user in localStorage. Just keep that in mind as the project evolves.